transformer/dockerfilegenerator/nodejsdockerfiletransformer.go

```diff
@@ -97,6 +101,46 @@ func (t *NodejsDockerfileGenerator) GetConfig() (transformertypes.Transformer, *
 	return t.Config, t.Env
 }
 
+// getNodeVersion returns the Node version to be used for the service
+func getNodeVersion(versionConstraint, defaultNodejsVersion string, supportedVersions []int) string {
+	v1, err := version.NewVersion(versionConstraint)
+	if err == nil {
+		logrus.Debugf("the constraint is a Node version: %#v", v1)
+		node := "v" + versionConstraint
+		version := strings.TrimPrefix(semver.Major(node), "v")
+		nodeVersion, err := strconv.Atoi(version)
+		if err != nil {
+			logrus.Errorf("unable to convert Node major version '%d' to int. Selecting default Node version- %s", nodeVersion, defaultNodejsVersion)
+			return defaultNodejsVersion
+		}
+		if nodeVersion%2 != 0 { // to make the version closest to the previous even version
+			nodeVersion = nodeVersion - 1
+		}
+		sort.Ints(supportedVersions)
+		if nodeVersion < supportedVersions[0] {
+			return strconv.Itoa(supportedVersions[0])
+		}
+		if nodeVersion > supportedVersions[len(supportedVersions)-1] {
+			return strconv.Itoa((supportedVersions[len(supportedVersions)-1]))
+		}
+		return strconv.Itoa(nodeVersion)
```

Comment:
Instead of relying on specific idea like even version, we can possibly find the nearest (greater?) version in the list.

Reply:
Now, we go over the versions in the list, and which ever version first satisfies the constraint we select it.